pkgs/by-name/xo/xonsh/wrapper.nix

```diff
@@ -0,0 +1,26 @@
+{
+  lib,
+  callPackage,
+  runCommand,
+  xonsh,
+  # configurable options
+  extraPackages ? (ps: [ ]),
+}:
+
+let
+  inherit (xonsh.passthru) python3;
+
+  pythonEnv = python3.withPackages
```

Comment:
Does this approach allow overriding the python version (not the packages), i.e. using xonsh with e.g. python 3.12 despite python 3.11.9 being the current python3 version?

Reply:
It uses the same Python provided by `xonsh.passthru`.

Reply:
ah so i can override python with e.g. `xonsh.override {python3 = pkgs.python312; }` and it'll eat it? Incredible, ty :)